AdvLoggerPkg/Library/AdvancedLoggerLib/DxeCore/AdvancedLoggerLib.c

```diff
@@ -384,8 +384,10 @@ DxeCoreAdvancedLoggerLibConstructor (
       LoggerInfo->LogBufferSize = EFI_PAGES_TO_SIZE (FixedPcdGet32 (PcdAdvancedLoggerPages)) - sizeof (ADVANCED_LOGGER_INFO);
       LoggerInfo->LogCurrent    = LoggerInfo->LogBuffer;
       LoggerInfo->HwPrintLevel  = FixedPcdGet32 (PcdAdvancedLoggerHdwPortDebugPrintErrorLevel);
-      mMaxAddress               = LoggerInfo->LogBuffer + LoggerInfo->LogBufferSize;
-      mBufferSize               = LoggerInfo->LogBufferSize;
+      AdvancedLoggerHdwPortInitialize ();

```

Comment:
Should this call be conditionalized on `!mLoggerInfo->HdwPortInitialized`?

The code obviously depends on `LoggerInfo` presence to implicitly determine if the port was initialized above but given that `mLoggerInfo->HdwPortInitialized` appears to be an explicit guard around the port being initialized more than once, it would be a bit more robust to guard the call with it regardless of other dependencies like the implicit side effect/return value from `AdvancedLoggerGetLoggerInfo()`.

Reply:
Well, the main reason I didn't do that is because LoggerInfo was just created and zero'd immediately prior, so there is no code path where LoggerInfo->HdwPortInitialized would be true - making the conditional redundant. I can add it if you think that is more self-describing though. Also - at this point mLoggerInfo is probably null, so the conditional would need to be on LoggerInfo anyway.

Reply:
Within the current context I agree it's not needed. But the flag really is associated with that call so I think it should always surround the call. People change code without being aware of all the things defined elsewhere, copy/paste calls elsewhere and don't realize associated flags otherwise existed, etc. So, this is more a "defensive programming" suggestion.

Reply:
Ok, added the conditional. Also fixed the mLoggerInfo reference which should have been a LoggerInfo reference.